Composite/Core/Localization/LocalizationFacade.cs

```diff
@@ -367,12 +367,26 @@ public static void RemoveLocale(CultureInfo cultureInfo, bool makeFlush = true)
                     UserSettings.RemoveActiveLocaleCultureInfo(username, cultureInfo);
                 }
 
+                IEnumerable<IUserGroupActiveLocale> userGroupsLocales = DataFacade.GetData<IUserGroupActiveLocale>();
+                foreach(IUserGroupActiveLocale el in userGroupsLocales)
+                {
+                    if (el.CultureName == cultureName)
+                    {
+                        DataFacade.Delete(el);
+                    }
+                }
+

```

Comment:
There's a small nuance here. When deleting in a loop, it is important to evaluate the list of elements to be deleted beforehand. The reasoning is - when using an SQL data provider, if one iterates a query, and then deletes an row from it, the next iteration of the for loop will fail with an SQL exception.

```
var userGroupsLocales = DataFacade.GetData<IUserGroupActiveLocale>().Where(l => l.CultureName == cultureName).ToList();
DataFacade.Delete(userGroupsLocales.Cast<IData>());
```


However, in this particular case, there's an easier and a more declarative way of implementing it in C1. You can mark a data type property as ForeignKey, and point to another field in another data type. When the referenced item is deleted, this one will be deleted as well.


	public interface IUserGroupActiveLocale : IData
	{

    // ....
	[ForeignKey(typeof(ISystemActiveLocale), nameof(ISystemActiveLocale.CultureName), AllowCascadeDeletes = true)]
	string CultureName { get; set; }

Reply:
Dmitry, thank you a lot, fantastic comment, will keep in mind. The pull request was updated with ForeignKey.